docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,303 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Clustering enables Grafana Agent Flow to coordinate a fleet of agents to
+work together for workload distribution and high availability. It makes agent
+deployments horizontally scalable by default, and minimizes the operational
+overhead of managing your agent infrastructure to handle demand.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model and
+the lightweight [grafana/ckit][] framework. All running nodes are assumed to
+use the same configuration and have access to the same type of hardware and
+network resources such as PVCs and service discovery APIs.
+
+Clustering is built from the ground-up to be the de-facto answer to "how do I
+scale my Grafana Agent setup" in the future. As an example, in comparison to a
+setup using [hashmod sharding][], clustering with target auto-distribution
+scales and provides resiliency at _half_ the cost in resources and provides
+dynamic resharding without the need to change configuration files.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+1-node cluster.
+
+## Usage
+
+The following set of command-line flags are used to configure clustering:
+- `--server.http.listen-addr`: the address where the agent’s HTTP server
+  listens to.
+- `--cluster.enabled`: enables cluster awareness.
+- `--cluster.node-name`: defines the name used by the cluster node. If not
+  provided, it defaults to the machine's hostname.
+- `--cluster.advertise-address`: defines the address the agent advertises for
+  its peers to connect to. If not provided, it is inferred automatically.
+- `--cluster.join-addresses`: accepts a comma-separated list of addresses to
+  join the cluster at. These can be IP addresses with an optional port, or a
+DNS record to lookup.
+
+Peers communicate over HTTP/2 on the agent's built-in HTTP server. Each node
+must be configured to accept connections from one another on the address
+defined or inferred in `--cluster.advertise-address`.
+
+Each cluster member’s name must be unique within the cluster. Nodes which try
+to join with a conflicting name are rejected.
+
+If the advertised address is not explicitly set, the agent tries to infer a
+suitable one from the `eth0` and `en0` local network interfaces.
```

Comment:
This is missing information on:

* What does the inference do? 
* What happens when the inference fails?

Reply:
I've added some more details when moving this back to the run command page. Should we put more detail?